cpp/collections_module/algorithms/max.hpp

```diff
@@ -0,0 +1,33 @@
+#include <list>
+
+#include <mgp.hpp>
+
+constexpr std::string_view kResultMax = "max";
+
+void Max(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    const auto &list = arguments[0].ValueList();
+
+    if (list.Size() == 0) {
+      throw mgp::ValueException("Empty input list.");
+    }
+
+    mgp::Value max = mgp::Value(list[0]);
```

Comment:
does `list[0]` already return mgp::Value? If it does you can change this in the following way:
`mgp::Value &max = ....`